Migrate Admin pages to React

kravets-levko opened the pull request:
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature

## Description

- [x] Migrate System Status page
- [x] Refine Celery Status page
- [x] Migrate Outdated Queries page

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

![image](https://user-images.githubusercontent.com/12139186/54191940-93c34100-44bf-11e9-925b-b39a2f1fcd0d.png)

![image](https://user-images.githubusercontent.com/12139186/54192001-ae95b580-44bf-11e9-8a64-6629236c9d4a.png)

![image](https://user-images.githubusercontent.com/12139186/54192065-d1c06500-44bf-11e9-9bbd-d4e1313181d6.png)

ranbena commented:
Wow 1st look, looks so good! 👏
Great use of flexbox.

Few things I noticed:
1. System Status > Queues > Celery doesn't display a tooltip on hover.
2. Outdated Queries > "Last updated" value is "few seconds" on master, and "49 years ago" in this.
3. Outdated Queries - when there are no queries, no need to show the sidebar and the "Sorry, we couldn't find anything." isn't a good fit for this situation.

kravets-levko commented:
@ranbena Thanks! All fixed.

arikfr commented on client/app/components/admin/CeleryStatus.jsx:
```diff
@@ -46,7 +46,14 @@ function parseTasks(tasks) {
+      columns={columns}
+      rowKey="name"
+      dataSource={queues}
+      loading={{ spinning: loading, indicator: <i className="fa fa-spinner fa-pulse" /> }}
```
This is definitely better than Ant's default (ugly) indicator :-)

Can we use [`Spin.setDefaultIndicator`](https://ant.design/components/spin/#Static-Method) to globally set this?

arikfr commented on client/app/components/admin/Layout.jsx:
```diff
@@ -0,0 +1,39 @@
+      <PageHeader title="Admin" />
+
+      <div className="bg-white tiled">
+        <ul className="tab-nav">
```
Can't we use Ant's tabs here?

kravets-levko replied:
If tab header can be a link and have an empty body - yes. I'll check

arikfr commented on client/app/components/admin/StatusBlock.jsx:
```diff
@@ -0,0 +1,118 @@
+                className="m-l-5"
+                title={isString(queue.data_sources) && (queue.data_sources !== '') ? queue.data_sources : 'n/a'}
+                mouseLeaveDelay={0}
+              >
```
The Tooltip is not needed -- we no longer send the data sources list.

kravets-levko replied:
On preview instance it works :thinking_face:

arikfr replied:
Because it doesn't run the latest version of the code :) Try locally.

arikfr commented:
![image](https://user-images.githubusercontent.com/71468/54183030-c19e8a80-44ab-11e9-8dd2-3e9c66c88491.png)

DB size was humanized before.

ranbena commented:
Suggestion for future - make these into Redash built-in dashboards (grids, counts). 🤘

ranbena commented on client/app/components/admin/Layout.jsx:
```diff
@@ -0,0 +1,38 @@
+export default function Layout({ activeTab, children }) {
+  return (
+    <div className="container admin-page-layout">
+      <PageHeader title="Admin" />
```
ux nit: Shouldn't this be titled "System Status"?

kravets-levko replied:
No, "System Status" is the first tab

kravets-levko commented:
@ranbena That idea crossed my mind some time ago :slightly_smiling_face: We'll need some new visualization types for that.

ranbena commented on client/app/components/admin/CeleryStatus.jsx:
```diff
@@ -46,7 +46,14 @@ function parseTasks(tasks) {
 function QueuesTable({ loading, queues }) {
```
Perhaps an opportunity to fix the "forEach" error thrown when there're no celery tasks.
```js
function parseTasks(tasks = []) {
```

ranbena commented:
> @ranbena That idea crossed my mind some time ago 🙂 We'll need some new visualization types for that.

It could be a great forcing function to make better widgets and set an example to our clients.

arikfr commented:
> It could be a great forcing function to make better widgets and set an example to our clients.

I agree. Let's start moving in this direction when our visualizations are pure React, so it's easier to create better APIs for them. :-)

kravets-levko commented:
@ranbena (cc @arikfr) I fixed bug with `forEach` and refactored that page a bit more: sometimes it tried to call `setState` on unmounted component (I fixed that as well) and refined code - created more smaller components, and moved main logic to main (page) component.

kravets-levko commented:
@ranbena Which tabs do you mean? There are top-level tabs ("System Status", "Celery Status", ...) and inner tabs on "Celery Status". For top-level tabs I had to disable animation because that tabs are link, and when clicked - animation started, then component destroyed - so it looked really weird. For inner tabs animation doesn't look so weird.

ranbena commented:
> @ranbena Which tabs do you mean? 

Celery tabs

> For inner tabs animation doesn't look so weird.

Okey dokey

kravets-levko commented:
Decided to disable animation - for consistency between top-level and inner tabs.

kravets-levko commented:
@arikfr @gabrieldutra If you're fine with this - LMK if I can merge it. Thanks!
